src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,156 @@
+
+    public static void main(String[] args) {
+        Scanner scan = new Scanner(System.in);
+        String stdIn = "greet";
+        String stdOut = "";
+
+        do {
+            try {
+                stdOut = parseCmd(stdIn);
+            } catch (UnknownArgumentException e) {
+                stdOut = e.toString();
+            } catch (EmptyArgumentException e) {
+                stdOut = e.toString();
+            } catch (NumberFormatException e) {
+                stdOut = e.toString();
+            } catch (IndexOutOfBoundsException e) {
+                stdOut = e.toString();
+            } catch (Exception e) {
+                stdOut = e.toString();
+            } finally {
+                System.out.println(HR + stdOut + HR);
+            }
+
+            stdIn = scan.nextLine();
+        } while (!stdIn.equals("bye"));
+
+        System.out.println(BYE);
+    }
+
+    protected static String parseCmd(String line) throws EmptyArgumentException, UnknownArgumentException {
+        // Take commands from first space-separated word
+        String[] cmd = line.split(" ");
+        if (cmd.length == 0) {
+            throw new EmptyArgumentException("command");
+        }
+
+        switch (cmd[0]) {
+
+        case "greet": {
+            return GREET;
+        }
+
+        case "echo": {
+            String statement = joinUntil(cmd, 1, cmd.length);
+            return String.format(ECHO, statement);
+        }
+
+        case "list": {
+            StringBuilder out = new StringBuilder("     Here are the tasks in your list:\n");
+            for (int i = 1; i <= tasks.size(); ++i) {
+                out.append(String.format(TASK, i, tasks.get(i - 1)));
+            }
+            return out.toString();
+        }
+
+        case "done": {
+            int i = Integer.parseInt(cmd[1]);
+            tasks.get(i - 1).markAsDone();
+            return String.format(DONE, tasks.get(i - 1));
+        }
+
+        case "todo": {
+            if (cmd.length <= 1) {
+                throw new EmptyArgumentException("todo");
+            }
+            String description = joinUntil(cmd, 1, cmd.length);
+            Todo t = new Todo(description);
+            return addTask(t);
+        }
+
+        case "deadline": {
+            int i = indexOfString(cmd, "/by");
+            String description = joinUntil(cmd, 1, i);
+            String by = joinUntil(cmd, i + 1, cmd.length);
+            Deadline d = new Deadline(description, by);
+            return addTask(d);
+        }
+
+        case "event": {
+            int i = indexOfString(cmd, "/at");
+            String description = joinUntil(cmd, 1, i);
+            String at = joinUntil(cmd, i + 1, cmd.length);
+            Event e = new Event(description, at);
+            return addTask(e);
+        }
+
+        default: {
+            throw new UnknownArgumentException();
+        }
+        }
+    }
+
+    /**
+     * addTask adds task to array and prints total number of tasks
```

Comment:
For this comment here, you don't need to specify `addTask`. You can begin the sentence with `Adds task to...`.

Reply:
This also applies to your following functions `addTask, `indexOfString` etc.